src/heretic/g_game.c

```diff
@@ -780,6 +780,27 @@ void G_DoLoadLevel(void)
     // [crispy] update the "singleplayer" variable
     CheckCrispySingleplayer(!demorecording && !demoplayback && !netgame);
 
+    // [crispy] more ammo
```

Comment:
Is the `-autohealth` parameter considered demo-safe in Heretic and Hexen?

Reply:
@FozzeY Hello?

Reply:
Sorry, holidays! Turns out this parameter is not demo-safe - I have added a conditional for it, however, it is currently does not prevent demo recording due to #807.

Reply:
Seems we cannot go with a single `crispy->singleplayer` variable in Heretic and Hexen. Please remove the `CheckCrispySingleplayer()` line above and change the check in the `crispy->pistolstart` block from `crispy->singleplayer` to `!demorecording && !demoplayback && !netgame`.
Please move the additional checks for the extra parameters to the beginning of `D_DoomLoop()` and change the `!crispy->singleplayer` checks to `!demorecording && gameaction != ga_playdemo && !netgame`. This applies to both Heretic and Hexen.

Reply:
The issue is that at this point `demorecording` is still `false`, so we cannot rely on it either. Perhaps setting the `demorecording = true;` before the `G_InitNew` call in `G_RecordDemo` would be feasible?
https://github.com/fabiangreffrath/crispy-doom/blob/05b426763b6381a09178d4ac1e986f99d176452c/src/heretic/g_game.c#L2094-L2151

Reply:
That's why I asked you to "please move the additional checks for the extra parameters to the beginning of D_DoomLoop() and change the !crispy->singleplayer checks to !demorecording && gameaction != ga_playdemo && !netgame." I know it's confusing! :wink: